platform/darwin/src/MGLMultiPoint.mm

```diff
@@ -93,6 +84,47 @@ - (void)getCoordinates:(CLLocationCoordinate2D *)coords range:(NSRange)range
     }
 }
 
+- (void)setCoordinates:(CLLocationCoordinate2D *)coords range:(NSRange)range
+{
+    if ([self isMemberOfClass:[MGLMultiPoint class]])
+    {
+        [NSException raise:@"Unsupported method" format:@"The coordinates of a member of MGLMultiPoint are read-only."];
+    }
+
+    if (range.length == 0)
+    {
+        [[NSException exceptionWithName:NSRangeException
+                                 reason:[NSString stringWithFormat:@"Empty coordinate range %@", NSStringFromRange(range)]
+                               userInfo:nil] raise];
+    }
+    else if (range.location > _count)
+    {
+        [[NSException exceptionWithName:NSRangeException
+                                 reason:[NSString stringWithFormat:
+                                            @"Invalid range %@ for existing coordinate count %zu",
+                                            NSStringFromRange(range), _count]
+                               userInfo:nil] raise];
+    }
+
+    [self willChangeValueForKey:@"coordinates"];
+    if (NSMaxRange(range) <= _count)
+    {
+        // replacing existing coordinate(s)
+        memcpy(_coordinates + range.location, coords, range.length * sizeof(CLLocationCoordinate2D));
```

Comment:
This will catch the case in which the developer tries to close a polyline:

```objc
[polyline setCoordinates:polyline.coordinates range:NSMakeRange(polyline.count, 1)];
```

but not if they try to have it return to the midpoint:

```objc
[polyline setCoordinates:polyline.coordinates + polyline.pointCount / 2 range:NSMakeRange(polyline.count, 1)];
```

I think the check should be expanded to something like:

```objc
(coords >= _coordinates + range.location && coords < pointCount + NSMaxRange(range))
|| (coords + range.length >= pointCount + range.location && coords + range.length < pointCount + NSMaxRange(range))
```

By the way, in that case, it would be safe to call [`memmove()`](http://www.cplusplus.com/reference/cstring/memmove/) instead of `memcpy()`.